src/OrbitGgp/Client.cpp

```diff
@@ -212,23 +217,51 @@ Future<ErrorMessageOr<Account>> ClientImpl::GetDefaultAccountAsync() {
       });
 }
 
-Future<ErrorMessageOr<std::vector<SymbolDownloadInfo>>> ClientImpl::GetSymbolDownloadInfoAsync(
-    const std::vector<SymbolDownloadQuery>& symbol_download_queries) {
-  QStringList arguments{"crash-report", "download-symbols", "-s", "--show-url"};
-
-  for (const auto& query : symbol_download_queries) {
-    arguments.push_back("--module");
-    arguments.push_back(QString("%1/%2")
-                            .arg(QString::fromStdString(query.build_id))
-                            .arg(QString::fromStdString(query.module_name)));
-  }
+Future<ErrorMessageOr<NotFoundOr<SymbolDownloadInfo>>> ClientImpl::GetSymbolDownloadInfoAsync(
+    const SymbolDownloadQuery& symbol_download_query) {
+  QStringList arguments{"crash-report",
+                        "download-symbols",
+                        "-s",
+                        "--show-url",
+                        "--module",
+                        QString("%1/%2")
+                            .arg(QString::fromStdString(symbol_download_query.build_id))
+                            .arg(QString::fromStdString(symbol_download_query.module_name))};
 
   orbit_base::ImmediateExecutor executor;
   return orbit_qt_utils::ExecuteProcess(ggp_program_, arguments, this, absl::FromChrono(timeout_))
-      .ThenIfSuccess(&executor,
-                     [](const QByteArray& json) -> ErrorMessageOr<std::vector<SymbolDownloadInfo>> {
-                       return SymbolDownloadInfo::GetListFromJson(json);
-                     });
+      .Then(&executor,
+            [](ErrorMessageOr<QByteArray> call_ggp_result)
+                -> ErrorMessageOr<NotFoundOr<SymbolDownloadInfo>> {
+              if (call_ggp_result.has_error()) {
+                // When symbols are not found or some other errors occur (e.g., invalid input for
+                // the ggp call), orbit_qt_utils::ExecuteProcess returns the error message in the
+                // format of "Error occurred while executing process \"<process_description>\",
+                // error: <error>,\nstdout:\n<standard_output>\nstderr:\n<standard_error>\n".
+                QString error_msg = QString::fromStdString(call_ggp_result.error().message());
+                QRegularExpression errorRegex("stderr:\n(.+)\n");
+                QRegularExpressionMatch errorMatch = errorRegex.match(QString(error_msg));
+
+                if (!errorMatch.hasMatch()) return ErrorMessage{call_ggp_result.error().message()};
```

Comment:
I am torn about failing harder here. We control how the error message that contains `stderr` is constructed. And I think if `call_ggp_result.has_error()`, then it really *always* contains `stderr`. 

So my initial thought is: This should be a `ORBIT_CHECK(errorMatch.hasMatch())`. But then on the other hand that is also not a great idea. So maybe a compromise is to log ORBIT_ERROR in this case? @beckerhe wdyt?